packages/website/src/shared/ui/date-input/date-input.tsx

```diff
@@ -1,22 +1,55 @@
-import { PropsWithChildren, useState } from "react";
+import clsx from "clsx";
+import { PropsWithChildren, useEffect, useState } from "react";
 import Calendar from "@lucky-parking/ui/src/components/calendar";
 import { formatToMiddleEndian } from "@/shared/lib/utilities/date";
+import { useMultiRefs } from "@/shared/lib/utilities/useMultiRef";
 
 export default function DateInput({ children }: PropsWithChildren) {
+  const [multiRefs, addMultiRef] = useMultiRefs();
   const [value, setValue] = useState<string | null>();
   const [isCalendarVisible, setCalendarVisible] = useState(false);
 
+  useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      const arr = multiRefs() as HTMLElement[];
+      if (!arr.length) return;
+
+      //check if click event is within any of our refs
```

Comment:
I understand what we're trying to accomplish here which is good, but are there any other approaches we can do?

Reply:
Good catch. I did a refactor and found a better solution. I remove the add multi Refs and just refactored picklist to render in the body of the calendar. That way we don't need to use multiple refs.

https://github.com/hackforla/lucky-parking/pull/560/commits/ef10b01f7b636a5cf46a68ee7e24139d73ee28bf

One thing I think we need to do soon, is create a custom class for our z-indexes